entc/integration/integration_test.go

```diff
@@ -2870,6 +2871,36 @@ func OrderByEdgeTerms(t *testing.T, client *ent.Client) {
 	}
 }
 
+func QueryM2MWithAdditionalFields(t *testing.T, client *ent.Client) {
+	require := require.New(t)
+
+	ctx := context.Background()
+	usr := client.User.Create().SetName("foo").SetAge(20).SaveX(ctx)
+	client.User.Query().OnlyX(ctx)
+
+	inf := client.GroupInfo.Create().SetDesc("desc").SaveX(ctx)
+	client.Group.Create().SetName("Github").SetExpire(time.Now()).AddUsers(usr).SetInfo(inf).SaveX(ctx)
+	client.Group.Query().OnlyX(ctx)
+
+	const key = "computed_value"
+	users := client.User.Query().WithGroups(func(query *ent.GroupQuery) {
+		query.Where(func(selector *sql.Selector) {
+			selector.AppendSelectExprAs(sql.Expr("min(3, 5)"), key)
+		})
+	}).AllX(ctx)
```

Comment:
If you need to modify the query selection, use the [query modifiers](https://entgo.io/docs/feature-flags/#select-and-scan-dynamic-values). e.g.,
```suggestion
		query.Modify(func(s *sql.Selector) {
			s.AppendSelectExprAs(sql.Expr("min(3, 5)"), key)
		})
	}).AllX(ctx)
```
This should solve your issue. Please, try update the test and see if this works without changing the codegen.